charts/base-cluster/templates/backup/validation.yaml

```diff
@@ -3,3 +3,9 @@
     {{- fail (printf "The `defaultLocation` '%s' must exist in `backupStorageLocations`" .Values.backup.defaultLocation) -}}
   {{- end -}}
 {{- end -}}
+{{- if .Values.backup.enabled -}}
+  {{- $length := len .Values.backup.backupStorageLocations -}}
+  {{- if eq $length 0 -}}
+    {{- fail (print "At least one .Values.backup.backupstoragelocations needs to be configured")  -}}
+  {{- end -}}
+{{- end -}}
```

Comment:
You should replace this whole block with the following inside the schema;

```diff
diff --git a/charts/base-cluster/values.schema.json b/charts/base-cluster/values.schema.json
index 18c64f5..62b367b 100644
--- a/charts/base-cluster/values.schema.json
+++ b/charts/base-cluster/values.schema.json
@@ -1278,7 +1278,8 @@
               "bucket"
             ],
             "additionalProperties": false
-          }
+          },
+          "minProperties": 1
         },
         "defaultLocation": {
           "type": "string"
@@ -1293,6 +1294,9 @@
           "additionalProperties": false
         }
       },
+      "required": [
+        "defaultLocation"
+      ],
       "additionalProperties": false
     },
     "common": {
```

which should accomplish the same

Reply:
no longer necessary